store/errors.go

```diff
@@ -73,6 +73,9 @@ type RevisionNotAvailableError struct {
 }
 
 func (e *RevisionNotAvailableError) Error() string {
+	if e.Channel != "" {
+		return fmt.Sprintf("no snap revision available in %q", e.Channel)
```

Comment:
we don't know here in general if the problem is the channel or the architecture, we need slightly complicated code to find that out (see SnapRevisionNotAvailable in daemon.go),  either way it means it problably should be its own branch

Reply:
Yeah, we should probably drop that form this PR for now.